thirdparty/misc/ifaddrs-android.h

```diff
@@ -32,15 +32,22 @@
 // Fills out a list of ifaddr structs (see below) which contain information
 // about every network interface available on the host.
 // See 'man getifaddrs' on Linux or OS X (nb: it is not a POSIX function).
+extern "C" {
 struct ifaddrs {
   struct ifaddrs* ifa_next;
   char* ifa_name;
   unsigned int ifa_flags;
   struct sockaddr* ifa_addr;
   struct sockaddr* ifa_netmask;
-  // Real ifaddrs has broadcast, point to point and data members.
-  // We don't need them (yet?).
+  // We never initialize the following members. We only define them to match the ifaddrs struct.
+  union
+  {
+    struct sockaddr *ifu_broadaddr;
+    struct sockaddr *ifu_dstaddr;
+  } ifa_ifu;
+  void *ifa_data;
```

Comment:
I needed to add the `ifaddrs` missing members as the Mono BCL will do a copy of the struct so it must match. Nothing else changes, these members are still never initialized (the Mono BCL doesn't use them either so that's fine).
This is a third-party header, should this change's diff be documented somewhere?

Reply:
It would be good if you can include a patch in `thirdparty/misc/patches/` that does those changes, and mention it where relevant in `thirdparty/README.md`.

Reply:
Alternatively, you can enclose all Godot-made changes with
```
// -- GODOT start --
...
// -- GODOT end --
```
so that we can easily identify what we added ourselves.

Reply:
Which one is the preferred way?

Reply:
I did it the start/end comments way.